programs/uxd/src/state/mercurial_vault_depository.rs

```diff
@@ -81,9 +85,33 @@ pub struct MercurialVaultDepository {
     pub redeemable_amount_under_management_cap: u128,
 
     pub minting_disabled: bool,
+
+    // Total amount of interests collected by interests_and_fees_redeem_authority
+    pub profits_total_collected: u128,
+
+    // Worth 0 if interests and fees never got collected
+    pub last_profits_collection_unix_timestamp: u64,
+
+    // Receiver of the depository's profits
+    pub profits_beneficiary_collateral: Pubkey,
 }
 
 impl MercurialVaultDepository {
+    pub fn update_onchain_accounting_following_profits_collection(
+        &mut self,
+        collected_profits: u64,
+        current_time_as_unix_timestamp: u64,
+    ) -> std::result::Result<(), UxdError> {
```

Comment:
nit: don't need to use std::result here, you can use the Anchor one to keep codebase more consistent